repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -3585,6 +3478,26 @@ public List<String> propagateClassification(String entityGuid, String classifica
             AtlasVertex currentTaskVertex = (AtlasVertex) graph.query().has(TASK_GUID, currentTask.getGuid()).vertices().iterator().next();
             currentTaskVertex.setProperty(TASK_ASSET_COUNT_TO_PROPAGATE, impactedVertices.size());
 
+
+            //iterate over each impacted vertex to create a message and send it to kafka
+            for (AtlasVertex vertex : impactedVertices) {
+                // Create a Map to store vertex properties
+                Map<String, Object> vertexMap = new HashMap<>();
+                ObjectMapper objectMapper = new ObjectMapper(); // Jackson ObjectMapper
+
+                vertexMap.put("parentTaskVertexId", vertex.getProperty("parentTaskVertexId", String.class));
+                vertexMap.put("action", "ADD_PROPAGATION");
+                vertexMap.put("assetVertexId", vertex.getIdForDisplay());
+                vertexMap.put("tagVertexId", classificationVertexId);
+                vertexMap.put("parentTaskGuid", currentTask.getGuid());
+                vertexMap.put("assetGuid", vertex.getIdForDisplay());
```

Comment:
Let's remove `assetGuid` from the payload. It's the same as `assetVertexId`. It's redundant.

Reply:
Removed